### PR TITLE
Add job endpoint ACL

### PR DESCRIPTION
**Pull request by chelseakomlo**

### CODE REVIEW

File: website/source/api/jobs.html.md [hunk 1]
```diff
@@ -972,9 +972,9 @@ The table below shows this endpoint's support for
 [blocking queries](/api/index.html#blocking-queries) and
 [required ACLs](/api/index.html#acls).
 
-| Blocking Queries | ACL Required |
-| ---------------- | ------------ |
-| `YES`            | `none`       |
+| Blocking Queries | ACL Required               |
+| ---------------- | -------------------------- |
```
Comment (dadgar) [inline website/source/api/jobs.html.md]: Extra width?

File: nomad/job_endpoint_test.go [hunk 2]
```diff
@@ -2006,6 +2007,89 @@ func TestJobEndpoint_GetJobSummary(t *testing.T) {
...
+		WriteRequest: structs.WriteRequest{
+			Region:    "global",
+			Namespace: job.Namespace,
+		},
+	}
+	reg.SecretID = root.SecretID
+
+	// Register the job with a valid token
+	var regResp structs.JobRegisterResponse
+	if err := msgpackrpc.CallWithCodec(codec, "Job.Register", reg, &regResp); err != nil {
```
Comment (dadgar) [inline nomad/job_endpoint_test.go]: Can you use assert on new tests

File: nomad/job_endpoint_test.go [hunk 3]
```diff
@@ -2006,6 +2007,89 @@ func TestJobEndpoint_GetJobSummary(t *testing.T) {
...
+	validToken := CreatePolicyAndToken(t, state, 1001, "test-valid",
+		NamespacePolicy(structs.DefaultNamespace, "", []string{acl.NamespaceCapabilityReadJob}))
+
+	req.SecretID = validToken.SecretID
+	var authResp structs.JobSummaryResponse
+	if err := msgpackrpc.CallWithCodec(codec, "Job.Summary", req, &authResp); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	expectedJobSummary := &structs.JobSummary{
```
Comment (dadgar) [inline nomad/job_endpoint_test.go]: Can you assert successful with management token (`root`)

File: website/source/api/jobs.html.md (hunk 1 above)
Comment (chelseakomlo) [reply website/source/api/jobs.html.md]: The line below adds more width so this is to bring all lines to the same width.

**Pull request merged**